apps/console/src/features/applications/api/application.ts

```diff
@@ -984,3 +984,80 @@ export const getRequestPathAuthenticators = (): Promise<any> => {
                 error.config);
         });
 };
+
+/**
+ * Function to add/update My Account portal status.
+ *
+ * @param status - My Account portal status.
+ *
+ * @returns Promise of response of the My Account status update request.
+ * @throws IdentityAppsApiException
+ */
+ export const updateMyAccountStatus = (status: boolean): Promise<any> => {
+
+    const config = `{"name": "status", "attributes": [{"key": "enable", "value": ${ status }}]}`;
+
+    const requestConfig: AxiosRequestConfig = {
+        data: config,
+        headers: {
+            "Accept": "application/json",
+            "Access-Control-Allow-Origin": store.getState().config.deployment.clientHost,
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.PUT,
+        url: store.getState().config.endpoints.myAccountConfigMgt
+    };
+
+    return httpClient(requestConfig)
+        .then((response: AxiosResponse) => {
+            if (response.status !== 200) {
+                throw new IdentityAppsApiException(
+                    ApplicationManagementConstants.MYACCOUNT_STATUS_UPDATE_INVALID_STATUS_CODE_ERROR,
+                    null,
+                    response.status,
+                    response.request,
+                    response,
+                    response.config);
+            }
+
+            return Promise.resolve(response.data);
+        })
+        .catch((error: AxiosError) => {
+            throw new IdentityAppsApiException(
+                ApplicationManagementConstants.MYACCOUNT_STATUS_UPDATE_ERROR,
+                error.stack,
+                error.code,
+                error.request,
+                error.response,
+                error.config);
+        });
+};
+
+/**
+ * Hook to get the status of the My Account Portal.
+ *
+ * @returns Reponse of the My Account status retrieval request.
+ */
+export const getMyAccountStatus = <Data = Object, Error = RequestErrorInterface>(
```

Comment:
`Object` type is generic. Better create an interface.

```suggestion
export const getMyAccountStatus = <Data = Object, Error = RequestErrorInterface>(
```

Reply:
Addressed with 06eac07bbcb1c736313953e0cc52b62e0ac383a7